### PR TITLE
feat: add api keys

### PR DESCRIPTION
**Pull request by manuel-rw**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

### CODE REVIEW

File: apps/nextjs/src/app/[locale]/manage/tools/api/components/api-keys.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  const columns = useMemo<MRT_ColumnDef<RouterOutputs["apiKeys"]["getAll"][number]>[]>(
+    () => [
+      {
+        accessorKey: "id",
+        header: "ID",
+      },
+      {
+        accessorKey: "user",
+        header: "Created by",
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/[locale]/manage/tools/api/components/api-keys.tsx:44]: Add translations for table headers

File: apps/nextjs/src/app/[locale]/manage/tools/api/components/api-keys.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  const table = useMantineReactTable({
+    columns,
+    data: apiKeys,
+    renderTopToolbarCustomActions: () => (
+      <Button
+        onClick={() => {
+          mutate();
+        }}
+        disabled={isPending}
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/[locale]/manage/tools/api/components/api-keys.tsx:64]: Disabled is unnecessary as it's automatically applied when loading is true

File: apps/nextjs/src/app/[locale]/manage/tools/api/components/swagger-ui.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+interface SwaggerUIClientProps {
+  document: OpenAPIV3.Document;
+}
+
+export const SwaggerUIClient = ({ document }: SwaggerUIClientProps) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const requestInterceptor = (req: Record<string, any>) => {
+    req.credentials = "omit";
+    return req;
+  };
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/[locale]/manage/tools/api/components/swagger-ui.tsx:20]: Is this necessary?

File: apps/nextjs/src/app/[locale]/manage/tools/api/page.tsx [hunk 4]
```diff
@@ -21,8 +19,25 @@ export async function generateMetadata() {
   };
 }
 
-export default function ApiPage() {
+export default async function ApiPage() {
   const document = openApiDocument(extractBaseUrlFromHeaders(headers()));
-
-  return <SwaggerUI spec={document} />;
+  const apiKeys = await api.apiKeys.getAll();
+  const t = useScopedI18n("management.page.tool.api.tab");
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/[locale]/manage/tools/api/page.tsx:25]: This one should be await getScopedI18n, right?

File: apps/nextjs/src/app/api/[...trpc]/route.ts [hunk 5]
```diff
@@ -1,14 +1,53 @@
...
 
 import { appRouter, createTRPCContext } from "@homarr/api";
+import type { Session } from "@homarr/auth";
+import { createSessionAsync } from "@homarr/auth/server";
+import { db, eq } from "@homarr/db";
+import { apiKeys } from "@homarr/db/schema/sqlite";
+import { logger } from "@homarr/log";
+
+const handlerAsync = async (req: Request) => {
+  logger.warn(JSON.stringify(req.headers));
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/api/[...trpc]/route.ts:11]: I guess this one has to be removed

File: apps/nextjs/src/app/api/[...trpc]/route.ts [hunk 6]
```diff
@@ -1,14 +1,53 @@
...
+          },
+        },
+      },
+    });
+
+    if (apiKeyFromDb !== undefined) {
+      logger.info(`Read session from API request and found user ${apiKeyFromDb.user.name} (${apiKeyFromDb.user.id})`);
+      session = await createSessionAsync(db, apiKeyFromDb.user);
+    }
+  }
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/api/[...trpc]/route.ts:43]: extract to seperate method to make code more readable

File: packages/api/src/router/app.ts [hunk 7]
```diff
@@ -7,53 +7,114 @@ import { validation, z } from "@homarr/validation";
...
+  byId: publicProcedure
+    .input(validation.app.byId)
+    .output(
+      z.object({
+        name: z.string(),
+        id: z.string(),
+        description: z.string().nullable(),
+        iconUrl: z.string(),
+        href: z.string().nullable(),
+      }),
```
Comment (Meierschlumpf) [inline packages/api/src/router/app.ts:88]: We'll probably be able to generate zod schemas from the db tables

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

File: packages/api/src/router/app.ts (hunk 7 above)
Comment (manuel-rw) [reply packages/api/src/router/app.ts:88]: Do you have an example?

File: apps/nextjs/src/app/[locale]/manage/tools/api/components/swagger-ui.tsx (hunk 3 above)
Comment (Meierschlumpf) [reply apps/nextjs/src/app/[locale]/manage/tools/api/components/swagger-ui.tsx:20]: Makes sense!

File: packages/api/src/router/app.ts (hunk 7 above)
Comment (Meierschlumpf) [reply packages/api/src/router/app.ts:88]: https://orm.drizzle.team/docs/zod

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

**Review by Meierschlumpf — APPROVED** [review]

**Pull request merged**